From benh@kernel.crashing.org Thu Sep 21 19:57:36 2006
From: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Subject: powerpc: Fix ohare IDE irq workaround on old powermacs
To: Greg KH <greg@kroah.com>
Date: Fri, 22 Sep 2006 12:57:26 +1000
Message-Id: <1158893846.26347.147.camel@localhost.localdomain>

From: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Looks like a workaround for old bogus OF bitrot... This fixes it and hence fixes
boot on some performa machines.

Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/ide/ppc/pmac.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.18.orig/drivers/ide/ppc/pmac.c
+++ linux-2.6.18/drivers/ide/ppc/pmac.c
@@ -1326,7 +1326,7 @@ pmac_ide_macio_attach(struct macio_dev *
 	if (macio_irq_count(mdev) == 0) {
 		printk(KERN_WARNING "ide%d: no intrs for device %s, using 13\n",
 			i, mdev->ofdev.node->full_name);
-		irq = 13;
+		irq = irq_create_mapping(NULL, 13);
 	} else
 		irq = macio_irq(mdev, 0);
 
